tools/apprunner-common/src/main/java/org/projectnessie/quarkus/runner/ListenUrlWaiter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.runner;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+import java.util.function.LongSupplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Accepts {@link String}s via it's {@link #accept(String)} method and checks for the
+ * {@code Listening on: http...} pattern.
+ */
+final class ListenUrlWaiter implements Consumer<String> {
```

Comment:
Presumably this means if the Quarkus log line changes or the user swiches logging levels we would never pick up the started prcoess? Will the started process still be killed or oprhaned?

Reply:
If it doesn't find the pattern, it will kill the process and error out.